File: evd_authoring_app/src/model/evdScript/program/tasks/openGripper.js
import { Task } from '../task';
import { Gripper } from '../primitives';
import { NodeParser } from '../../utilityFunctions';


export class OpenGripper extends Task {

    /*
    * Data structure methods
    */

    static typeString() {
        return 'open-gripper';
    }

    static fullTypeString() {
        return Task.fullTypeString() + OpenGripper.typeString();
    }

    constructor(thingUuid=null, position=0, effort=100, speed=100, type='', name='', 
                uuid=null, parent=null, appendType=true, primitives=null) 
    {

        if (primitives !== null) {
            primitives = [
                Gripper(
                    thingUuid,
                    position,
                    effort,
                    speed
                )
            ];
        }

        super(
            primitives,
            (appendType) ? 'open-gripper.'+type : type,
            name,
            uuid,
            parent,
            appendType
        );
    }

    static fromDict(dct) {
        return new OpenGripper(
            null,
            0,
            100,
            100,
            dct.type,
            dct.name,
            dct.uuid,
            null,
            false,
            dct.primitives.map(p => NodeParser(p))
        );
    }
}

File: evd_ros_backend/evd_ros_core/src/evd_script/program/skills/open_gripper.py
'''
Simplified parameterization for gripper primitive to fully open
'''

from ..skill import Skill
from ..primitives import Gripper


class OpenGripper(Skill):

    '''
    Data structure methods
    '''

    @classmethod
    def type_string(cls, trailing_delim=True):
        return 'open-gripper' + ('.' if trailing_delim else '')

    @classmethod
    def full_type_string(cls):
        return Skill.full_type_string() + cls.type_string()

    def __init__(self, position=0, effort=100, speed=100, thing_uuid=None, type='',
                 name='', uuid=None, parent=None, append_type=True, primitives=None, editable=True, deleteable=True):

        if primitives == None:
            primitives=[
                Gripper(
                    thing_uuid=thing_uuid,
                    semantic=Gripper.SEMANTIC_RELEASING if thing_uuid != None else Gripper.SEMANTIC_AMBIGUOUS,
                    position=position,
                    effort=effort,
                    speed=speed,
                    editable=editable,
                    deleteable=deleteable)
            ]

        super(OpenGripper,self).__init__(
            type=OpenGripper.type_string() + type if append_type else type,
            name=name,
            uuid=uuid,
            parent=parent,
            append_type=append_type,
            primitives=primitives,
            editable=editable,
            deleteable=deleteable)
